### PR TITLE
Change agedFailoverTimeMultiplier config to doPriorityLeaseTaking

### PR DESCRIPTION
**Pull request by lucienlu-aws**
*Issue #, if available:*

*Description of changes:*

Change the agedFailoverTimeMultiplier config to doPriorityLeaseTaking. This allows clients to disable the priority lease taking logic entirely if they choose

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/MultiLangDaemonConfiguration.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ public void setWorkerId(String workerId) {
     @ConfigurationSettable(configurationClass = LeaseManagementConfig.class)
     private long failoverTimeMillis;
     @ConfigurationSettable(configurationClass = LeaseManagementConfig.class)
-    private int agedFailoverTimeMultiplier;
+    private Boolean doPriorityLeaseTaking;
```
Comment (lucienlu-aws) [inline amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/MultiLangDaemonConfiguration.java:90]: After doing some testing, I found that when this value is a primitive `boolean`, the config isn't set properly. This applies to all other boolean configs in multilang as well. Will make another PR fixing it for other configs

File: amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseManagementConfig.java [hunk 2]
```diff
@@ -104,13 +104,14 @@ public class LeaseManagementConfig {
     private long failoverTimeMillis = 10000L;
 
     /**
-     * Multiplier for the failoverTimeMillis in which leases which are expired for an extended period of time defined by
-     * (agedFailoverTimeMultiplier * failoverTimeMillis) are taken with priority, disregarding the target
-     * but obeying the maximum limit per worker.
+     * Whether workers should take very expired leases at priority. A very expired lease is when a worker does not
+     * renew its lease in 3 * {@link LeaseManagementConfig#failoverTimeMillis}. Very expired leases will be taken at
```
Comment (akidambisrinivasan) [inline amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseManagementConfig.java:108]: Can you also mentioned newly created leases due to shard mutation that are not assigned to any worker are also considered very-expired ?

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseTakerTest.java [hunk 3]
```diff
@@ -151,6 +152,32 @@ public void test_veryOldLeaseDurationNanosMultiplierGetsCorrectLeases() throws E
...
+    public void test_disableDoPriorityLeaseTakingGetsCorrectLeases() throws Exception {
+        long veryOldThreshold = MOCK_CURRENT_TIME -
+                (TimeUnit.MILLISECONDS.toNanos(LEASE_DURATION_MILLIS) * DEFAULT_VERY_OLD_LEASE_DURATION_MULTIPLIER);
+        DynamoDBLeaseTaker dynamoDBLeaseTakerWithCustomMultiplier =
+                new DynamoDBLeaseTaker(leaseRefresher, WORKER_IDENTIFIER, LEASE_DURATION_MILLIS, metricsFactory)
+                        .withDoPriorityLeaseTaking(false);
+        final List<Lease> allLeases = new ArrayList<>();
+        allLeases.add(createLease("foo", "2", MOCK_CURRENT_TIME));
+        allLeases.add(createLease("bar", "3", veryOldThreshold - 1));
+        allLeases.add(createLease("baz", "4", veryOldThreshold + 1));
```
Comment (akidambisrinivasan) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseTakerTest.java:165]: Also add some leases not assigned to any worker?

File: amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseManagementConfig.java [hunk 4]
```diff
@@ -57,7 +57,7 @@ public class LeaseManagementConfig {
     public static final long DEFAULT_COMPLETED_LEASE_CLEANUP_INTERVAL_MILLIS = Duration.ofMinutes(5).toMillis();
     public static final long DEFAULT_GARBAGE_LEASE_CLEANUP_INTERVAL_MILLIS = Duration.ofMinutes(30).toMillis();
     public static final long DEFAULT_PERIODIC_SHARD_SYNC_INTERVAL_MILLIS = 2 * 60 * 1000L;
-    public static final int DEFAULT_AGED_FAILOVER_TIME_MULTIPLIER = 3;
+    public static final boolean DEFAULT_DO_PRIORITY_LEASE_TAKING = true;
```
[the remark this answers is not in the document]
Comment (lucienlu-aws) [reply amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseManagementConfig.java:60]: I like that naming better, updated

File: amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseManagementConfig.java (hunk 2 above)
Comment (lucienlu-aws) [reply amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseManagementConfig.java:108]: Updated

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseTakerTest.java (hunk 3 above)
Comment (lucienlu-aws) [reply amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseTakerTest.java:165]: Updated

**Review by akidambisrinivasan — APPROVED** [review]
